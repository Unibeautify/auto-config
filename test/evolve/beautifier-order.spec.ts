import { testGenetic } from "../testGenetic";
import prettier from "@unibeautify/beautifier-prettier";
import jsBeautify from "@unibeautify/beautifier-js-beautify";

testGenetic("beautifier-order", {
  beautifiers: [prettier, jsBeautify],
  language: "JavaScript",
  originalText: `const fun1 = function (arg1) {
}
const fun2 = function(arg1) {
}

var foo = {
    bar: "baz",
    qux: "quux",
};

var bar = {
    bar: "baz",
    qux: "quux"
};

`,
  desiredText: `/** @format */

const fun1 = function ( arg1 ) {}
const fun2 = function ( arg1 ) {}

var foo = {
    bar: "baz"
    , qux: "quux"
, }

var bar = {
    bar: "baz"
    , qux: "quux"
, }
`,
});
