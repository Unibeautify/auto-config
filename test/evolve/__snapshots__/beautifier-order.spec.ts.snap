// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`beautifier-order: beautifiedText 1`] = `
/** @format */

const fun1 = function ( arg1 ) {}
const fun2 = function ( arg1 ) {}

var foo = {
    bar: "baz"
    , qux: "quux"
, }

var bar = {
    bar: "baz"
    , qux: "quux"
, }

`;

exports[`beautifier-order: entity 1`] = `
Object {
  "options": Object {
    "beautifiers": Array [
      "Prettier",
      "JS-Beautify",
    ],
    "comma_first": true,
    "end_with_comma": true,
    "end_with_newline": true,
    "end_with_semicolon": false,
    "jslint_happy": true,
    "pragma_insert": true,
    "space_in_paren": true,
  },
}
`;

exports[`beautifier-order: fitness 1`] = `1006`;
